src/Adapter/Order/CommandHandler/DeleteProductFromOrderHandler.php

```diff
@@ -26,41 +26,71 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Order\CommandHandler;
 
+use Cart;
+use CartRule;
 use Configuration;
+use Currency;
+use Customer;
 use Hook;
 use Order;
 use OrderCarrier;
+use OrderCartRule;
 use OrderDetail;
 use OrderInvoice;
+use PrestaShop\PrestaShop\Adapter\ContextStateManager;
 use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
 use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Order\Product\Command\DeleteProductFromOrderCommand;
 use PrestaShop\PrestaShop\Core\Domain\Order\Product\CommandHandler\DeleteProductFromOrderHandlerInterface;
 use PrestaShop\PrestaShop\Core\Domain\Order\ValueObject\OrderId;
+use PrestaShop\PrestaShop\Core\Localization\CLDR\ComputingPrecision;
+use Tools;
 use Validate;
 
 /**
  * @internal
  */
 final class DeleteProductFromOrderHandler extends AbstractOrderCommandHandler implements DeleteProductFromOrderHandlerInterface
 {
+    /**
+     * @var ContextStateManager
+     */
+    private $contextStateManager;
+
+    /**
+     * @param ContextStateManager $contextStateManager
+     */
+    public function __construct(ContextStateManager $contextStateManager)
+    {
+        $this->contextStateManager = $contextStateManager;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function handle(DeleteProductFromOrderCommand $command)
     {
         $orderDetail = new OrderDetail($command->getOrderDetailId());
         $order = new Order($command->getOrderId()->getValue());
+        dump($order);
 
         $this->assertProductCanBeDeleted($order, $orderDetail);
 
+        $cart = new Cart($order->id_cart);
+
+        $this->contextStateManager
```

Comment:
Maybe restore the context if an exception is thrown

Reply:
Fixed